--- src/runtime/Config.x86-64-win32     2024-04-29 16:18:52.000000000 -0500
+++ src/runtime/Config.x86-64-win32     2024-05-12 13:26:58.119261982 -0500
@@ -53,10 +53,11 @@
 # -Wno-type-limits suppresses a ton of warnings from 'grovel-headers'
 # which seems to produce C code in which a comparison is always false.
 CFLAGS += -g -W -Wall \
-       -Wno-unused-function -Wno-unused-parameter -Wno-cast-function-type \
+       -Wno-unused-function -Wno-unused-parameter \
        -Wno-type-limits \
        -fno-omit-frame-pointer \
        -O5 -m64 -DWINVER=0x0501 \
+       -I"${BUILD_PREFIX}\include" \
        -D__W32API_USE_DLLIMPORT__

 CC = gcc
