--- src/runtime/Config.x86-64-linux     2024-04-29 16:18:52.000000000 -0500
+++ src/runtime/Config.x86-64-linux     2024-05-27 18:47:57.623786724 -0500
@@ -37,14 +37,14 @@
 endif

 ifdef LISP_FEATURE_SB_CORE_COMPRESSION
-  OS_LIBS += -lzstd
+  OS_LIBS += -L${PREFIX}/lib -lzstd
 endif

 ifdef HAVE_LIBUNWIND
   OS_LIBS += -lunwind
 endif

-CFLAGS += -Wunused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer
+CFLAGS += -Wunused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer
 DISABLE_PIE=no

 ifdef LISP_FEATURE_IMMOBILE_SPACE
