--- clean.sh    2024-04-29 16:18:51.000000000 -0500
+++ clean.sh    2024-05-11 19:26:44.737874895 -0500
@@ -114,4 +114,4 @@
         -name 'depend' -o \
         -name 'TAGS' -o \
         -name 'tags' -o \
-        -name 'local-target-features.lisp-expr' \) -print | xargs rm -fr
+        -name 'local-target-features.lisp-expr' \) -print | env -i /usr/bin/xargs /usr/bin/rm -fr
--- src/runtime/Config.x86-64-win32     2024-04-29 16:18:52.000000000 -0500
+++ src/runtime/Config.x86-64-win32     2024-05-12 13:26:58.119261982 -0500
@@ -38,7 +38,7 @@

 OS_LIBS = -l ws2_32 -ladvapi32 -lm
 ifdef LISP_FEATURE_SB_CORE_COMPRESSION
-  OS_LIBS += -lzstd
+  OS_LIBS += -L${BUILD_PREFIX}\\Library\\ucrt64\\lib -lzstd
 endif
 ifdef LISP_FEATURE_SB_FUTEX
   OS_LIBS += -lSynchronization
@@ -59,2 +59,3 @@
        -O5 -m64 -DWINVER=0x0501 \
+       -I${BUILD_PREFIX}\\Library\\ucrt64\\include -I${PREFIX}\\ucrt64\\include \
        -D__W32API_USE_DLLIMPORT__
@@ -62,2 +62,2 @@

-CC = gcc
+# CC = ${CC}
--- make-shared-library.sh      2024-05-30 02:25:20.000000000 -0500
+++ make-shared-library.sh      2024-06-18 18:35:29.173493526 -0500
@@ -7,2 +7,2 @@

-$GNUMAKE -C src/runtime libsbcl.so
+$GNUMAKE -C src/runtime libsbcl.dll
