--- make-config.sh      2024-04-29 16:18:51.000000000 -0500
+++ make-config.sh      2024-05-26 10:16:49.402352909 -0500
@@ -779,6 +779,7 @@
     else
         $GNUMAKE -C tools-for-build determine-endianness -I ../src/runtime
         "$CROSSCOMPILING_EMULATOR" tools-for-build/determine-endianness >> $ltf
+        echo ' :os-provides-dlopen' >> $ltf
     fi
     export sbcl_os sbcl_arch android
     sh tools-for-build/grovel-features.sh >> $ltf
